File: backend/src/main/java/com/devrats/model/WebhookPayload.java
package com.devrats.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public record WebhookPayload(
        @JsonProperty("ref") String ref,
        @JsonProperty("commits") List<CommitData> commits,
        @JsonProperty("sender") SenderData sender,
        @JsonProperty("repository") RepoData repository,
        @JsonProperty("installation") InstallationRef installation
) {
    @JsonIgnoreProperties(ignoreUnknown = true)
    public record CommitData(
            @JsonProperty("id") String id,
            @JsonProperty("message") String message,
            @JsonProperty("timestamp") String timestamp,
            @JsonProperty("author") AuthorData author,
            @JsonProperty("added") List<String> added,
            @JsonProperty("removed") List<String> removed,
            @JsonProperty("modified") List<String> modified
    ) {}

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record AuthorData(
            @JsonProperty("name") String name, 
            @JsonProperty("email") String email, 
            @JsonProperty("username") String username
    ) {}

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record SenderData(
            @JsonProperty("id") Long id, 
            @JsonProperty("login") String login, 
            @JsonProperty("avatar_url") String avatar_url
    ) {}

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record RepoData(
            @JsonProperty("id") Long id, 
            @JsonProperty("name") String name, 
            @JsonProperty("full_name") String full_name, 
            @JsonProperty("private") Boolean private_repo
    ) {}

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record InstallationRef(
            @JsonProperty("id") Long id
    ) {}
}

File: backend/src/main/kotlin/com/devrats/features/github/models/WebhookPayload.kt
package com.devrats.features.github.models

import kotlinx.serialization.Serializable

@Serializable
data class WebhookPayload(
    val ref: String? = null,
    val commits: List<CommitData>? = null,
    val sender: SenderData? = null,
    val repository: RepoData? = null,
    val installation: InstallationRef? = null
)

@Serializable
data class CommitData(
    val id: String,
    val message: String,
    val timestamp: String,
    val author: AuthorData? = null,
    val added: List<String> = emptyList(),
    val removed: List<String> = emptyList(),
    val modified: List<String> = emptyList()
)

@Serializable
data class AuthorData(val name: String? = null, val email: String? = null, val username: String? = null)

@Serializable
data class SenderData(val id: Long, val login: String, val avatar_url: String? = null)

@Serializable
data class RepoData(val id: Long, val name: String, val full_name: String, val private: Boolean = false)

@Serializable
data class InstallationRef(val id: Long)
